### PR TITLE
Add script to compare poses between two reconstructions in colmap format

### PR DESCRIPTION
**Pull request by ayushbaid**
Checked locally, pose metrics look good.

### CODE REVIEW

File: gtsfm/evaluation/compare_colmap_outputs.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    for wTi, fname in zip(wTi_list, image_fnames):
+        if wTi is None or fname is None:
+            continue
+
+        poses[fname] = wTi
+
+    return poses
+
+
+def compare_poses(baseline_dirpath: str, current_dirpath: str, output_path: str):
```
Comment (johnwlambert) [inline gtsfm/evaluation/compare_colmap_outputs.py:33]: nit: missing return type

could we also add an `Args` section describing the arguments here?

File: gtsfm/evaluation/compare_colmap_outputs.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    runner_base.save_metrics_reports([ba_pose_metrics], metrics_path=output_path)
+
+
+if __name__ == "__main__":
+    """
+    Compare two reconstructions (in Colmap's output format). Right now, we just compare the poses.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--baseline",
```
Comment (johnwlambert) [inline gtsfm/evaluation/compare_colmap_outputs.py:79]: could we use the term "reference" instead, if this is the target / reference?

File: gtsfm/evaluation/compare_colmap_outputs.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+import gtsfm.runner.gtsfm_runner_base as runner_base
+import gtsfm.utils.geometry_comparisons as geometry_comparisons
+import gtsfm.utils.io as io_utils
+import gtsfm.utils.metrics as metric_utils
+from gtsfm.evaluation.metrics import GtsfmMetricsGroup
+
+
+def load_poses(colmap_dirpath: str) -> Dict[str, Pose3]:
+    wTi_list, image_fnames = io_utils.read_images_txt(fpath=os.path.join(colmap_dirpath, "images.txt"))
```
[the remark this answers is not in the document]
Comment (johnwlambert) [reply gtsfm/evaluation/compare_colmap_outputs.py:21]: Ref: https://github.com/borglab/gtsfm/blob/master/gtsfm/visualization/view_scene.py#L46C1-L48C6

File: gtsfm/evaluation/compare_colmap_outputs.py (hunk 2 above)
Comment (travisdriver) [reply gtsfm/evaluation/compare_colmap_outputs.py:79]: I like baseline

File: gtsfm/evaluation/compare_colmap_outputs.py (hunk 1 above)
Comment (travisdriver) [reply gtsfm/evaluation/compare_colmap_outputs.py:33]: +1

File: gtsfm/evaluation/compare_colmap_outputs.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (travisdriver) [reply gtsfm/evaluation/compare_colmap_outputs.py:21]: +1

**Review by travisdriver — APPROVED** [review]
LGTM except few small comments

File: gtsfm/evaluation/compare_colmap_outputs.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ayushbaid) [reply gtsfm/evaluation/compare_colmap_outputs.py:21]: Done

[the remark this answers is not in the document]
Comment (johnwlambert) [reply gtsfm/evaluation/compare_colmap_outputs.py]: Thanks!

[the remark this answers is not in the document]
Comment (johnwlambert) [reply gtsfm/evaluation/compare_colmap_outputs.py]: (marking as resolved).

File: gtsfm/evaluation/compare_colmap_outputs.py (hunk 1 above)
Comment (johnwlambert) [reply gtsfm/evaluation/compare_colmap_outputs.py]: Thanks, marking as resolved.

**Pull request merged**